core/frontend/src/RealityDataSourceTilesetUrlImpl.ts

```diff
@@ -105,9 +110,17 @@ export class RealityDataSourceTilesetUrlImpl implements RealityDataSource {
     return true;
   }
 
-  /** Returns the tile URL. If the tile path is a full URL, it is returned as is. Otherwise, the base URL is prepended to the tile path. */
+  /** Returns the tile URL. If the tile path is a full URL, the URL's search parameters are checked.
+   * If the search parameters are empty, the root url's search parameters will be appended to the tile path.
+   * Otherwise, the base URL is prepended to the tile path, and the root url's search parameters are appended,
+   * if no search parameters exist on the existing tile path.
+   */
   private getTileUrl(tilePath: string){
-    return this.isValidURL(tilePath) ? tilePath : this._baseUrl + tilePath;
+    if (this.isValidURL(tilePath)) {
+      const url = new URL(tilePath);
+      return url.search === "" ? tilePath + this._searchParams : tilePath;
+    }
+    return tilePath.includes("?") ? this._baseUrl + tilePath : this._baseUrl + tilePath + this._searchParams;
```

Comment:
nit:
maybe use templates strings instead of string concat?
feel free to ignore & resolve on your own

Reply:
Agreed, much cleaner. Done in latest commit.